flang/lib/Lower/Intrinsics.cpp

```diff
@@ -693,6 +703,22 @@ mlir::Value IntrinsicLibrary::genAimag(mlir::Type resultType,
       args[0], true /* isImagPart */);
 }
 
+// ANINT
+mlir::Value IntrinsicLibrary::genAnint(mlir::Type resultType,
+                                       llvm::ArrayRef<mlir::Value> args) {
+  assert(args.size() >= 1);
+  // Skip optional kind argument to search the runtime
```

Comment:
Isn't the kind parameter semantically significant?

Reply:
It is, it determines the result type kind (and therefore has to be a compile time constant). Semantics analysis already applied that in the type of the expression.
So, if present, the kind is already reflected in the `resultType` that is used to search the runtime.
I added `it is already reflected in result type` to the comment.